dmutils/documents.py

```diff
@@ -171,5 +175,19 @@ def get_signed_url(bucket, path, base_url):
         return url
 
 
-def get_agreement_document_path(framework_slug, supplier_id, document_name):
-    return '{0}/agreements/{1}/{1}-{2}'.format(framework_slug, supplier_id, document_name)
+def get_agreement_document_path(framework_slug, supplier_id, supplier_name, document_name):
+    return '{0}/agreements/{1}/{2}-{1}-{3}'.format(
+        framework_slug,
+        supplier_id,
+        _sanitise_supplier_name(supplier_name),
+        document_name
+    )
+
+
+def _sanitise_supplier_name(supplier_name):
```

Comment:
I feel like this shouldn't be private, it's useful in it's own right at least in scripts.